### PR TITLE
Added Progress Bars

### PR DESCRIPTION
**Pull request by DonIsaac**
~~Requires #194 to be merged first~~ 194 is merged

- feat(scrapers): Added progress bars to all scrapers
- chore: added `dev:db` command to connect to DB with psql shell
- fix(scrapers/sections): Fixed bug causing abnormally long scrape times
- up(scrapers): Finished converting the rest of the scrapers into classes
- tests: Wrote `"a scraper"` behavior in `spec_helper`
- tests: Added `"a scraper"` behavior test cases to all scrapers
- ~~todo(scrapers): Create rspec behavior for scrapers (e.g. `it_has_behavior 'scraper'`)~~ done
- ~~todo(scrapers): Add test coverage for the rest of the scrapers (common and majors already have tests)~~ done(ish)
- ~~todo: Add documentation either to wiki, a separate `CONTRIBUTING.md`, or to the Wiki about how to create new scrapers~~done(ish) in a different PR

### CODE REVIEW

**Comment by DonIsaac** [comment]
This PR is ready for review 👍

**Comment by McIntireEvan** [comment]
I skimmed it and it seems pretty good. One thing before I approve - can you post sample output, with the progress bar? I'm worried about how it will interact with the logger, and how readable logs will remain after it, e.g. looking back at a scrape that auto-ran. (I can also run it on my machine later, but it helps completeness of the PR to have it explicitly here as well)

**Comment by DonIsaac** [comment]
@McIntireEvan they look kind of like this
image

the `log` command that I "wrote" is taken from `ruby-progressbar`'s `#log` function. However, their `log` function is equivalent to `puts`, so for those sweet, sweet timestamps I re-used the clear/refresh logic they used

File: app/scrapers/courses_scraper.rb [hunk 1]
```diff
@@ -5,181 +5,199 @@
...
+    # add the courses from each department to the database
+    dept_urls.each do |url|
+      dept_id = url.split('/soc/')[1][7, 10]
+      semester = url.split('/soc/')[1][0, 6]
+      courses = []
+      table_name = 'courses'
+
+      log(bar, :debug) { "Getting courses for #{dept_id} (#{semester})" }
+      # log(bar,)
+      # logger.debug(prog_name) { "fetching #{url}" }
```
Comment (tybug) [inline app/scrapers/courses_scraper.rb:71]: remove commented code?

File: app/scrapers/majors_scraper.rb [hunk 2]
```diff
@@ -41,29 +43,33 @@ def scrape_page(page)
...
 
   # @param [Array<Major>]
   # @return [void]
   def update_db(majors)
+    bar = get_progress_bar(title: "#{self.class} - Loading Majors", total: majors.length)
     $DB[:majors].delete
     majors.each do |major|
-      logger.info { "inserting #{major[:name]}" }
+      log(bar, :debug) { "inserting #{major[:name]}" }
+      # logger.debug { "inserting #{major[:name]}" }
```
Comment (tybug) [inline app/scrapers/majors_scraper.rb:65]: same here

File: app/scrapers/map_scraper.rb [hunk 3]
```diff
@@ -6,26 +6,34 @@
...
 
-logger = ScraperCommon.logger
-url = 'https://raw.githubusercontent.com/umdio/umdio-data/master/umd-building-gis.json'
+class MapScraper
+  include ScraperCommon
 
-url = 'https://raw.githubusercontent.com/umdio/umdio-data/master/umd-building-gis.json'
+  def url
+    'https://raw.githubusercontent.com/umdio/umdio-data/master/umd-building-gis.json'
+  end
```
Comment (tybug) [inline app/scrapers/map_scraper.rb:18]: why not a static method instead?

**Review by tybug — CHANGES_REQUESTED** [review]

File: app/scrapers/map_scraper.rb (hunk 3 above)
Comment (DonIsaac) [reply app/scrapers/map_scraper.rb:18]: Made it easier to refactor ¯\_(ツ)_/¯ the scrapers could use some more cleanup as URL handling isn't super consistent, some things still need to be broken into helper functions, etc. but that's cleanup for another PR IMO.

**Review by tybug — APPROVED** [review]

**Pull request merged**